File: src/main/kotlin/uk/nhs/nhsdigital/pmir/transforms/XPNtoHumanName.java
package uk.nhs.nhsdigital.pmir.transforms;

import ca.uhn.hl7v2.model.v24.datatype.XPN;
import org.apache.commons.collections4.Transformer;
import org.hl7.fhir.r4.model.HumanName;


public class XPNtoHumanName implements Transformer<XPN, HumanName> {

    @Override
    public HumanName transform(XPN xpn) {
        HumanName name = new HumanName();
        if (xpn.getGivenName() != null) {
            name.addGiven(xpn.getGivenName().getValue());
        }
        if (xpn.getFamilyName() != null) {
            name.setFamily(xpn.getFamilyName().getSurname().getValue());
        }
        if (xpn.getXpn5_PrefixEgDR() != null) {
            name.addPrefix(xpn.getXpn5_PrefixEgDR().getValue());
        }
        return name;
    }
}

File: src/main/kotlin/uk/nhs/england/tie/transforms/v24/XPNtoHumanName.kt
package uk.nhs.england.tie.transforms.v24

import ca.uhn.hl7v2.model.v24.datatype.XPN
import org.apache.commons.collections4.Transformer
import org.hl7.fhir.r4.model.HumanName


class XPNtoHumanName : Transformer<XPN?, HumanName?> {
    override fun transform(xpn: XPN?): HumanName {
        val name: HumanName = HumanName()
        if (xpn != null) {
            if (xpn.givenName != null) {
                name.addGiven(xpn.givenName.value)
            }
            if (xpn.familyName != null) {
                name.setFamily(xpn.familyName.surname.value)
            }
            if (xpn.xpn5_PrefixEgDR != null) {
                name.addPrefix(xpn.xpn5_PrefixEgDR.value)
            }
        }
        return name
    }
}
